vizro-ai/src/vizro_ai/plot2/_response_models.py

```diff
@@ -0,0 +1,218 @@
+                    f"Produced code execution failed the following error: <{e}>. Please check the code and try again."
+                )
+            assert isinstance(
+                fig, go.Figure
+            ), f"Expected chart code to return a plotly go.Figure object, but got {type(fig)}"
+            return v
+
+        return create_model(
+            "ChartPlanDynamic",
+            __validators__={
+                "validator1": validator("chart_code", allow_reuse=True)(_test_execute_chart_code),
+            },
+            __base__=ChartPlanStatic,
+        )
+
+
+if __name__ == "__main__":
+    # Ideas to implement - No code execution mode
+    # test OS models
+
+    from dotenv import find_dotenv, load_dotenv
+
+    load_dotenv(find_dotenv(usecwd=True))
+    # df = px.data.iris()
+    df = px.data.gapminder()
+
+    # dummy_model = _create_model(data_frame=df)
+    # dummy_model(chart_type="foo", imports=["bar"],chart_code="print('hello')",code_explanation="boo",chart_insights="bee")
+
+    # test = dummy_model(  # dummy_response = ChartPlanStatic(
+    #     chart_type="Bubble Chart",
+    #     imports=["import plotly.express as px"],
+    #     chart_code="def custom_chart(data_frame):\n    fig = px.scatter(data_frame, x='gdpPercap', y='lifeExp', size='pop', color='continent', hover_name='country', animation_frame='year', log_x=True, size_max=60)\n    return fig",
+    #     # chart_code="print('Hello World')\nx = 1",
+    #     chart_insights="This bubble chart visualizes the changes in life expectancy and GDP per capita over time. Each bubble represents a country, with the size of the bubble indicating the population and the color representing the continent. The animation is based on the year, showing the evolution of the data over time.",
+    #     code_explanation="1. Create a scatter plot using Plotly Express (px).\n2. Set the x-axis to 'gdpPercap' and the y-axis to 'lifeExp'.\n3. Size the bubbles based on 'pop' (population) and color them based on 'continent'.\n4. Enable hover information to display the country name.\n5. Add animation based on the 'year' column to animate the chart over time.\n6. Set log scaling for the x-axis.\n7. Return the Plotly figure object.",
+    # )
+
+    # model = _get_llm_model()
+
+    # query = "the trend of gdp over years in the US"
+    # query = "show me the geo distribution of life expectancy and set year as animation "
+    # query = "describe the composition of gdp in continents in 2007, and add horizontal line for avg gdp in 2007"
+    # query = "plot a bubble chart to shows the changes in life expectancy gdp per capita  over time. Animate the chart by year"
+
+    # print(df.sample(10).to_string())
+
+    # res = _get_pydantic_model(query=query, llm_model=model, response_model=dummy_model, df_info=df.sample(10).to_string())
+    # code = res._get_complete_code(lint=True)
+    # fig = res._get_fig_object(data_frame=df)
+    # fig.show()
+
+    # llm = ChatAnthropic(
```

Comment:
btw, I tested with Anthropic and it works very well 👏 you just need to ensure the base url is correct. One of these 3 did the trick:
```
print(os.environ.get("ANTHROPIC_API_URL"))
print(os.environ.get("ANTHROPIC_BASE_URL"))
print(os.environ.get("ANTHROPIC_API_BASE"))
```

Reply:
I had very good experiences with anthropic, that is really good to know